server/item/firework_explosion.go

```diff
@@ -0,0 +1,51 @@
+package item
+
+// FireworkExplosion represents an explosion of a firework.
+type FireworkExplosion struct {
+	// Shape represents the shape of the explosion.
+	Shape FireworkShape
+	// Colour is the colour of the explosion.
+	Colour Colour
+	// Fade is the colour the explosion should fade into.
```

Comment:
Might want to document that upon setting this, `Fades` must be set to true. Usually I'd be more of a fan to check against the default value to decide if it was changed, but obviously here that isn't possible. Documenting that would be useful though.

Reply:
Documented.